### ISSUE TITLE
Nested type descriptions

### ISSUE THREAD

**Issue by neutraali** [reactjs/react-docgen#683]
Sorry if this has been reported before in some form before, but is there currently support for nested type descriptions? For example, given the following structure (using [Flow](https://flow.org/en/) and `react-docgen@5.3.1`):

```js
type Panel = {
	/** Panel title */
	title: string,
	/** Panel content */
	content: string,
	/** Sublevels of more Panels */
	panels: Array<Panel>
};

type Props = {
	/** An accordien one or more have toggleable panels. */
	panels: Array<Panel>
}

/** Hello World! */
const Component = (props: Props) => { ... }
```
^ Does not generate any `description` for the **Panel** -props. Is this intentional? The output for a **Panel** is:

```
{
	"name": "signature",
	"type": "object",
	"raw": "{ ... }",
	"signature": {
		"properties": [
			{
				"key": "title",
				"value": {
					"name": "string",
					"required": true
				}
			},
			{
				"key": "content",
				"value": {
					"name": "string",
					"required": true
				}
			},
			{
				"key": "panels",
				"value": {
					"name": "Array",
					"elements": [{ "name": "Panel" }],
					"raw": "Array<Panel>",
					"required": true
				}
			}
		]
	}
}
```

**Comment by kevin-krug** [comment]
just would like to add my +1 here, in typescript nested interfaces seem to not be resolved currently.

E.g. the following parses the nested prop
```
interface Props {
  nestedProp: {
    nestedPropProp: string
  }
}

const Component = (props: Props) => { ... }
```

=>

```
[
  {
    "description": "",
    "displayName": "Component",
    "methods": [],
    "props": {
      "nestedProp": {
        "required": true,
        "tsType": {
          "name": "signature",
          "type": "object",
          "raw": "{\n  nestedPropProp: string\n}",
          "signature": {
            "properties": [
              {
                "key": "nestedPropProp",
                "value": {
                  "name": "string",
                  "required": true
                }
              }
            ]
          }
        },
        "description": ""
      }
    }
  }
]
```

VS.

whereas if the nested type references an interface, it is not resolved:

```
interface Props {
  nestedProp: NestedProp
}

interface NestedProp {
  nestedPropProp: string
}

const Component = (props: Props) => {
  return <div />;
}
```

=>

```
[
  {
    "description": "",
    "displayName": "Component",
    "methods": [],
    "props": {
      "nestedProp": {
        "required": true,
        "tsType": {
          "name": "NestedProp"
        },
        "description": ""
      }
    }
  }
]
```

**Comment by danez** [comment]
> just would like to add my +1 here, in typescript nested interfaces seem to not be resolved currently.
> 
> if this is not already supported, I'd request for the output of the `parse` function to include the nested props, also in case of nested interfaces. 🙏

When TypeScript support was implemented there was the decision made to handle interfaces different than object types and type aliases. The reason was that interfaces might be documented separately and linked from the react documentation. I can see that use case but also think that the same could apply for object types or enums.

What I'm thinking about is to handle interfaces and object types exactly the same, except to somehow make it clear that it is an interface. In addition it might be interesting to add a `referenceName`, so that in the end the name of the interface/typealias is still known.

Maybe something like this for your example above. Would that work for you?

```
"tsType": {
          "name": "signature",
          "referenceName" : "NestedProp", 
          "type": "interface",
          ...
}
```

**Comment by danez** [comment]
> Sorry if this has been reported before in some form before, but is there currently support for nested type descriptions? For example, given the following structure (using [Flow](https://flow.org/en/) and `react-docgen@5.3.1`):
> 
> ```js
> type Panel = {
> 	/** Panel title */
> 	title: string,
> 	/** Panel content */
> 	content: string,
> 	/** Sublevels of more Panels */
> 	panels: Array<Panel>
> };
> 
> type Props = {
> 	/** One or more toggleable Panels */
> 	panels: Array<Panel>
> }
> 
> /** Hello World! */
> const Component = (props: Props) => { ... }
> ```
> 
> ^ Does not generate any `description` for the **Panel** -props. Is this intentional? The output for a **Panel** is:
> 
> ```
> {
> 	"name": "signature",
> 	"type": "object",
> 	"raw": "{ ... }",
> 	"signature": {
> 		"properties": [
> 			{
> 				"key": "title",
> 				"value": {
> 					"name": "string",
> 					"required": true
> 				}
> 			},
> 			{
> 				"key": "content",
> 				"value": {
> 					"name": "string",
> 					"required": true
> 				}
> 			},
> 			{
> 				"key": "panels",
> 				"value": {
> 					"name": "Array",
> 					"elements": [{ "name": "Panel" }],
> 					"raw": "Array<Panel>",
> 					"required": true
> 				}
> 			}
> 		]
> 	}
> }
> ```

This issue with flow and missing descriptions will be fixed in the next release. see #810

**Comment by neutraali** [comment]
I don't think this is yet in the [latest release](https://github.com/reactjs/react-docgen/pull/937) (`7.1.0`). Will this be included in `7.2.0`? Is there an estimated timeline for the update?
